repos/fdbt-site/src/pages/api/managePassengerGroup.ts

```diff
@@ -0,0 +1,165 @@
+import { CompanionInfo, GroupPassengerType } from './../../interfaces/index';
+import { getAndValidateNoc, redirectTo } from './apiUtils/index';
+import { updateSessionAttribute } from './../../utils/sessions';
+import { isArray } from 'lodash';
+import { NextApiResponse } from 'next';
+import { ErrorInfo, NextApiRequestWithSession } from '../../../src/interfaces';
+import { redirectToError } from './apiUtils';
+import { GS_PASSENGER_GROUP_ATTRIBUTE } from '../../../src/constants/attributes';
+import { getPassengerTypeByNameAndNocCode, insertGroupPassengerType } from '../../data/auroradb';
+import { checkIntegerIsValid, removeExcessWhiteSpace } from './apiUtils/validator';
+
+export const formatRequestBody = (req: NextApiRequestWithSession): GroupPassengerType => {
+    const maxGroupSize = removeExcessWhiteSpace(req.body.maxGroupSize);
+    const groupName = removeExcessWhiteSpace(req.body.passengerGroupName);
+    const passengerTypes: string[] = !req.body.passengerTypes
+        ? []
+        : isArray(req.body.passengerTypes)
+        ? req.body.passengerTypes
+        : [req.body.passengerTypes];
+
+    const companions: CompanionInfo[] = [];
+    if (passengerTypes.length > 0) {
+        passengerTypes.forEach((passengerType) => {
```

Comment:
Minor: this could be done with a map to create the companions array from the passengerTypes array to save on the pushing and external initialisation of the empty array